File: ios/SAF/SAFModule.m
#import <React/RCTBridgeModule.h>

RCT_EXTERN_MODULE(SAFModule, NSObject)

RCT_EXTERN_METHOD(listFiles:(NSString *)uri
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(readFile:(NSString *)uri
                  encoding:(NSString *)encoding
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(exists:(NSString *)uri
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

File: ios/SAF/SAFModule.swift
@objc(SAF)
class SAFModule: NSObject {

    @objc static func requiresMainQueueSetup() -> Bool { return false }

    @objc func listFiles(_ uri: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        resolve([])
    }

    @objc func readFile(_ uri: String, encoding: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        resolve("")
    }

    @objc func exists(_ uri: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        resolve(false)
    }
}